consensus/propagation/have_wants.go

```diff
@@ -274,6 +276,28 @@ func (blockProp *Reactor) handleWants(peer p2p.ID, wants *proptypes.WantParts) {
 	}
 }
 
+// broadcastWants gossips the provided want msg to all peers except to the
+// original sender. This will be called when catching up.
+func (blockProp *Reactor) broadcastWants(wants *proptypes.WantParts, from p2p.ID) {
+	e := p2p.Envelope{
+		ChannelID: WantChannel,
+		Message: &propproto.WantParts{
+			Height: wants.Height,
+			Round:  wants.Round,
+			Parts:  wants.ToProto().Parts,
+		},
+	}
+	for _, peer := range blockProp.getPeers() {
+		if peer.peer.ID() == from {
+			continue
+		}
+
+		if !p2p.SendEnvelopeShim(peer.peer, e, blockProp.Logger) { //nolint:staticcheck
+			blockProp.Logger.Error("couldn't send want part", "target_peer", peer.peer.ID(), "height", wants.Height, "round", wants.Round)
+		}
+	}
+}
```

Comment:
this adds a feedback loop where we are downloading the block way more as we fall behind, therefore if a node gets congested, it ends up asking for more data, which is bad

Reply:
so should we limit the requests to 5 and then wait and then send another 5 for example?

Reply:
we should limit it to a low number, at least for concurrent requests. maybe increase to two concurrent requests if we realize that we're behind. avoiding the congestion feedback loop is super important unfortunately. 

The goal of pull based gossip during recovery should be to request enough data in the first place that this almost never happens.

this isn't strictly related to this code, but we should also first check those and request from nodes who have already sent us a have. 

If no peers have sent us haves but have sent us compact blocks for a future height, then that means they are also behind / getting eclipse attacked. We could probably push this case to handle later.